### DISCUSSION TITLE
Enum not working using wire:model and $wire.set

### DISCUSSION THREAD

**Discussion by sirbaconjr** [livewire/livewire#5575] [7. Bugs (archived)] [answered]
Native enums support was added in [this pull request](https://github.com/livewire/livewire/pull/4506). Unfortunately, although the middleware changed in that PR does its job, this fails when we get to `Livewire\HydrationMiddleware\PerformDataBindingUpdates` or `Livewire\HydrationMiddleware\PerformActionCalls`.

Am I doing something wrong? I feel like this was supposed to work.

**Comment by joshhanley** [comment, marked as answer]
@sirbaconjr Livewire supports enums being stored in properties. But Livewire currently doesn't know how to take a string update from the front end and convert it into an enum (this is the same with other property types too, like a carbon instance). 

So you will need to use an `updating` hook to validate the string and then convert it to an enum. The trick is you need to use value by reference (noticed the `&` in the method signature).

```php
public MyEnum $myEnum;

public function updatingMyEnum(&$value)
{
    // Validate $value is a valid enum string type
    
    $value = MyEnum::from($value);
}
```

Hope this helps!

**Comment by mjawad096** [comment]
@sirbaconjr You can use the updating hook with below code snippet to convert the updated value to enum for all enum type properties

```
public function updating($field, &$value)
{
    $type = (new ReflectionProperty($this, $field))->getType()?->getName();

    if(enum_exists($type)){
        $value = $type::from($value);
    }
}
```
